services/src/main/java/org/keycloak/services/resources/LoginActionsService.java

```diff
@@ -338,7 +342,28 @@ public Response processLogin(@QueryParam("code") String code,
                     .setUriInfo(uriInfo);
             return protocol.cancelLogin(clientSession);
         }
-
+        
+        UserModel user = KeycloakModelUtils.findUserByNameOrEmail(session, realm, username);
```

Comment:
@stianst check is introduced in LoginActionService itself so that the AuthenticationStatus changes to ACTIONS_REQUIRED (UPDATE_PASSWORD) and the control proceeds to nextActionAfterAuthentication. I think keeping this check in LoginActionService makes sense.